MiniAn/minian_main.py

```diff
@@ -0,0 +1,366 @@
+
+    # 3. Background removal
+    print(mn_defs.Messages.Main.PREPROC_REMOV_BACKG, flush=True)
+    with mn_utils.except_type_error("remove_background"):
+        varr_ref = remove_background(varr_ref, **minian_parameters.params_remove_background)
+
+    # Save
+    print(mn_defs.Messages.Main.PREPROC_SAVE, flush=True)
+    varr_ref = save_minian(varr_ref.rename("varr_ref"), intpath, overwrite=True)
+
+    return varr_ref
+
+def correct_motion(varr_ref, intpath, chk, minian_parameters):
+    ### Motion correction ###
+    if minian_parameters.parameters[defs.Parameters.danse.CORRECT_MOTION]:
+        print(mn_defs.Messages.Main.CORRECT_MOTION_ESTIM_SHIFT, flush=True)
+        with mn_utils.except_type_error("estimate_motion"):
+            motion = estimate_motion(varr_ref, **minian_parameters.params_estimate_motion)
+
+        motion = save_minian(motion.rename("motion").chunk({"frame": chk["frame"]}), **minian_parameters.params_save_minian)
+        print(mn_defs.Messages.Main.CORRECT_MOTION_APPLY_SHIFT, flush=True)
+        Y = apply_transform(varr_ref, motion, **minian_parameters.params_apply_transform)
+
+    else:
+        Y = varr_ref
+
+    print(mn_defs.Messages.Main.PREP_DATA_INIT, flush=True)
+    Y_fm_chk = save_minian(Y.astype(float).rename("Y_fm_chk"), intpath, overwrite=True)
+    Y_hw_chk = save_minian(Y_fm_chk.rename("Y_hw_chk"), intpath, overwrite=True,
+                           chunks={"frame": -1, "height": chk["height"], "width": chk["width"]})
+
+    return Y, Y_fm_chk, Y_hw_chk
+
+def initialize_seeds(Y_fm_chk, Y_hw_chk, minian_parameters):
+    ### Seed initialization ###
+    print(mn_defs.Messages.Main.INIT_SEEDS, flush=True)
+    with mn_utils.except_print_error_no_cells(mn_defs.Messages.Main.INIT_SEEDS):
+        # 1. Compute max projection
+        max_proj = save_minian(Y_fm_chk.max("frame").rename("max_proj"), **minian_parameters.params_save_minian).compute()
+        # 2. Generating over-complete set of seeds
+        with mn_utils.except_type_error("seeds_init"):
+            seeds = seeds_init(Y_fm_chk, **minian_parameters.params_seeds_init)
+
+        # 3. Peak-Noise-Ratio refine
+        print(mn_defs.Messages.Main.INIT_SEEDS_PNR_REFI, flush=True)
+        with mn_utils.except_type_error("pnr_refine"):
+            seeds, pnr, gmm = pnr_refine(Y_hw_chk, seeds, **minian_parameters.params_pnr_refine)
+
+        # 4. Kolmogorov-Smirnov refine
+        print(mn_defs.Messages.Main.INIT_SEEDS_KOLSM_REF, flush=True)
+        with mn_utils.except_type_error("ks_refine"):
+            seeds = ks_refine(Y_hw_chk, seeds, **minian_parameters.params_ks_refine)
+
+        # 5. Merge seeds
+        print(mn_defs.Messages.Main.INIT_SEEDS_MERG, flush=True)
+        seeds_final = seeds[seeds["mask_ks"] & seeds["mask_pnr"]].reset_index(drop=True)
+        with mn_utils.except_type_error("seeds_merge"):
+            seeds_final = seeds_merge(Y_hw_chk, max_proj, seeds_final, **minian_parameters.params_seeds_merge)
```

Comment:
`seeds_final` don't contain all the seeds, `seeds` has more rows than `seeds_final` 
You should combine 2 tables:
`df = pd.merge(seeds, seeds_final, how="outer").fillna(False)`